frontend/pages/booking/select-seat/index.tsx

```diff
@@ -1,11 +1,173 @@
+import { Seat } from '../../../components/Seat'
+import airplane from '../../../public/airplane.png';
+import { toast } from 'react-toastify'
+import { ISeat } from '@/interfaces/seat'
+const genRandomSold = () => {
+  const num = Math.random();
+  if (num < 0.3) return true;  //probability 0.3
+  else return false; // probability 0.7
+}
+
+const genRandomSeat = (): ISeat[] => {
+  const seats: ISeat[] = []
+  for (let i = 1; i < 7; i++) {
+    seats.push({
+      sold: genRandomSold(),
+      row: i,
+      column: "A"
+    })
+    seats.push({
+      sold: genRandomSold(),
+      row: i,
+      column: "B"
+    })
+    seats.push({
+      sold: genRandomSold(),
+      row: i,
+      column: "C"
+    })
+    seats.push({
+      sold: genRandomSold(),
+      row: i,
+      column: "D"
+    })
+  }
+  return seats
+}
 
 const SelectSeat:NextPage = () => {
+  const [allSeats, setAllSeats] = useState<ISeat[]>([])
+  useEffect(() => {
+
+    setAllSeats(genRandomSeat())
+  }, [])
+  const [selectedSeat, setSelectedSeat] = useState<ISeat | undefined>(undefined)
+  const router = useRouter();
+
+  const isSelected = (seat: ISeat): boolean => {
+    return seat.row === selectedSeat?.row && seat.column === selectedSeat?.column
+  }
+
+  const genSeatSelect = (seats: ISeat[]) => {
+    const seatA = seats.filter(seat => seat.column == "A")
+    const seatB = seats.filter(seat => seat.column == "B")
+    const seatC = seats.filter(seat => seat.column == "C")
+    const seatD = seats.filter(seat => seat.column == "D")
+
+    return seatA.map((sA: ISeat, index: number) => {
+      return (<>
+        <Grid item container justifyContent={"center"} alignItems={"center"} xs={1}>
+          <Typography variant={'h5'} component={'h4'} >
+            {index + 1}
+          </Typography>
+        </Grid>
+        <Grid item container xs={5}  >
+          <Grid item xs={6}>
+            <Seat seat={sA} selected={isSelected(sA)} changeSelectedSeat={changeSelectedSeat} />
+          </Grid>
+          <Grid item xs={6}>
+            <Seat seat={seatB[index]} selected={isSelected(seatB[index])} changeSelectedSeat={changeSelectedSeat} />
+          </Grid>
+        </Grid>
+        <Grid item container xs={1}>
+        </Grid>
+        <Grid item container xs={5} >
+          <Grid item xs={6}>
+            <Seat seat={seatC[index]} selected={isSelected(seatC[index])} changeSelectedSeat={changeSelectedSeat} />
+          </Grid>
+          <Grid item xs={6}>
+            <Seat seat={seatD[index]} selected={isSelected(seatD[index])} changeSelectedSeat={changeSelectedSeat} />
+          </Grid>
+        </Grid>
+      </>)
+    })
+  }
+
+  const changeSelectedSeat = (seat: ISeat) => {
+    if (seat.sold) {
+      return;
+    }
+    setSelectedSeat(seat)
+  }
+
+  const onNext = () => {
+    if (selectedSeat) {
+      router.push('/booking/select-inflight-service')
+    } else {
+      console.log('warn')
+      toast.warn("Please select a seat before moving on")
```

Comment:
please use snack bar in MUI (https://mui.com/material-ui/react-snackbar/#customized-snackbars) instead of mixing UI Library :).